### PR TITLE
add recursion limit to auto-gen

### PR DESCRIPTION
**Pull request by cmeeren**
See #19. Have focused on code, not style, so might need a bit of housekeeping before/after merge.

Please check if there are any important test cases I've missed.

### CODE REVIEW

File: tests/Hedgehog.Experimental.Tests/GenTests.fs [hunk 1]
```diff
@@ -3,6 +3,7 @@ module Hedgehog.Experimental.Tests.GenTests
 open Xunit
 open Swensen.Unquote
 open Hedgehog
+open Hedgehog
```
Comment (moodmosaic) [inline tests/Hedgehog.Experimental.Tests/GenTests.fs]: `open Hedgehog` twice?

File: src/Hedgehog.Experimental/Gen.fs [hunk 2]
```diff
@@ -228,17 +229,29 @@ module GenX =
           String = Gen.string (Range.linear 0 50) Gen.latin1
           DateTime = Gen.dateTime
           DateTimeOffset = Gen.dateTime |> Gen.map System.DateTimeOffset
-          SeqRange = Range.exponential 0 50 }
+          SeqRange = Range.exponential 0 50
+          RecursionDepth = 1 }
+
+    let rec private auto''<'a> (config : AutoGenConfig) (recursionDepths: Map<string, int>) : Gen<'a> =
```
Comment (moodmosaic) [inline src/Hedgehog.Experimental/Gen.fs]: Having *two* primes on a name, hmm...

Perhaps `autoWith`?

At this point I can tell that you prefer short names, but I think `auto''` might be confusing(?)

Comment (cmeeren) [reply src/Hedgehog.Experimental/Gen.fs]: It's a private function so I don't think it matters that much. But I like `autoWith`, I suggest we call the private function `autoInner`, and rename `auto'` (the public function that takes a config) to `autoWith`.

File: tests/Hedgehog.Experimental.Tests/GenTests.fs (hunk 1 above)
Comment (cmeeren) [reply tests/Hedgehog.Experimental.Tests/GenTests.fs]: That's just VF# fooling around. I'll remove it. :)

File: src/Hedgehog.Experimental/Gen.fs (hunk 2 above)
Comment (moodmosaic) [reply src/Hedgehog.Experimental/Gen.fs]: :+1:

File: tests/Hedgehog.Experimental.Tests/GenTests.fs (hunk 1 above)
Comment (moodmosaic) [reply tests/Hedgehog.Experimental.Tests/GenTests.fs]: :+1:

**Comment by cmeeren** [comment]
Okay, take a look :)

**Comment by moodmosaic** [comment]
I will, in the next couple of hours :)

**Comment by moodmosaic** [comment]
It really *is* time to setup Travis CI and release a NuGet package :tada:

**Pull request merged**